content/articles/spring-boot-recaptcha-v3/index.md

```diff
@@ -0,0 +1,855 @@
+
+### Create a SpringBoot application using spring initializr
+Go to [Spring Initialzr](https://start.spring.io/) and generate a new project with dependency `Spring Web`, `Spring Security`, `Spring Data JPA`, `Thymeleaf`, `Spring Boot Dev Tools`, `MySQL Driver` and `Lombok`.
+
+Import the generated project into Intellij and ensure you have an active internet connection to download the dependencies from the remote central repository.
+
+![Spring Initializr image](spring-initializr.png)
+
+### Configure database connection properties
+Since we will not deploy the application to production, the database tables, and data will be automatically created for us during application initialization using `spring.jpa.hibernate.ddl-auto=create` property.
+
+The SQL statements that will be generated will be displayed to the console to see what is happening behind the scenes and this will be enabled by `spring.jpa.show-sql=true` property.
+
+To ensure that we read the SQL statements easily, we can ensure they are formatted well by including the `spring.jpa.properties.hibernate.format_sql=true` property.
+
+application.properties
+```properties
+spring.datasource.url=jdbc:mysql://localhost:3306/recaptcha
+spring.datasource.username=username
+spring.datasource.password=password
+spring.jpa.hibernate.ddl-auto=create
+spring.jpa.show-sql=true
+spring.jpa.properties.hibernate.format_sql=true
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL8Dialect
+```
+
+### Create User, Role, and ReCaptchaResponse models
+We will make use of project `Lombok` to generate getters and setters, no-argument constructor, and all arguments constructor for all the models that will be used.
+
+Create a class named Role with fields `id` and `role name` where the role name represents the authority that a user can have.
+```java
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+import javax.persistence.*;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@ToString
+@Entity(name = "role")
+@Table
+public class Role {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    private String roleName;
+
+    public Role(String roleName) {
+        this.roleName = roleName;
+    }
+}
+
+```
+
+`@Data` - Generates all getters and setters for the fields in the class.
+
+`@AllArgsConstructor` - Generates a constructor with all the fields declared in the class.
+
+`@NoArgsConstructor` - Generates a constructor without any arguments.
+
+`@ToString` - Generates a string representation of the fields that can be used when debugging.
+
+
+Create a class named AppUser with fields `id`, `username`, and `password` which represents different users who can interact with the application depending on the rights they have.
+
+Since a user can have many rights we must include a Role field in the app user class and annotate with `@ManyToMany` annotation.
```

Comment:
Change to "Since users can have many rights, we must include a Role field in the `AppUser` class and annotate with the `@ManyToMany` annotation."